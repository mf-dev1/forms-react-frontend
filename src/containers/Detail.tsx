import * as React from 'react';

class Detail extends React.Component {
  public render() {
    return <h2 className="text-center">Detail</h2>;
  }
}

export default Detail;
