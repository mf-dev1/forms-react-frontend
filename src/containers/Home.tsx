import * as React from 'react';
import { connect } from 'react-redux';
import Typography from '@material-ui/core/Typography';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import FolderIcon from '@material-ui/icons/Folder';

export interface HomeItem {
  id: number;
  text: string;
}

export interface Props {
  items: HomeItem[];
  title: string;
}
class Home extends React.Component<Props> {
  public render() {
    const title = this.props.title;
    const items = this.props.items;

    const pageContent = (
      <List>
        {items.map((item, i) => (
          <ListItem>
            <ListItemIcon>
              <FolderIcon />
            </ListItemIcon>
            <ListItemText primary={item.text} secondary={null} />
          </ListItem>
        ))}
      </List>
    );
    return (
      <div className="home-container">
        <Typography variant="h6">{title}</Typography>
        {pageContent}
      </div>
    );
  }
}
// Map Redux state to component props
function mapStateToProps(state: any) {
  return {
    items: state.home.items,
    title: state.home.title
  };
}

// Connected Component
export default connect(
  mapStateToProps,
  null
)(Home);
