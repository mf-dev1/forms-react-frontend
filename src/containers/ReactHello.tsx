// src/components/StatefullHello.tsx

import * as React from 'react';
import Hello from '../components/Hello';
import '../styles/Hello.css';

export interface Props {
  name: string;
  enthusiasmLevel?: number;
}

interface State {
  currentEnthusiasm: number;
}

class ReactHello extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { currentEnthusiasm: props.enthusiasmLevel || 1 };
  }

  public onIncrement = () =>
    this._updateEnthusiasm(this.state.currentEnthusiasm + 1);
  public onDecrement = () =>
    this._updateEnthusiasm(this.state.currentEnthusiasm - 1);

  public render() {
    const { name } = this.props;
    return (
      <Hello
        name={name}
        enthusiasmLevel={this.state.currentEnthusiasm}
        onIncrement={this.onIncrement}
        onDecrement={this.onDecrement}
      />
    );
  }

  private _updateEnthusiasm(currentEnthusiasm: number) {
    this.setState({ currentEnthusiasm });
  }
}

export default ReactHello;
