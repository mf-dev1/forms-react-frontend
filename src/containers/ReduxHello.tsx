import { connect } from 'react-redux';
import Hello from '../components/Hello';
import '../styles/Hello.css';
import * as actions from '../actions';

// Map Redux state to component props
function mapStateToProps(state: any) {
  return {
    enthusiasmLevel: state.enthusiasmLevel,
    name: 'Redux',
  };
}

// Map Redux actions to component props
function mapDispatchToProps(dispatch: any) {
  return {
    onIncrement: () => dispatch(actions.increaseEnthusiasm()),
    onDecrement: () => dispatch(actions.decreaseEnthusiasm()),
  };
}

// Connected Component
const ReduxHello = connect(
  mapStateToProps,
  mapDispatchToProps
)(Hello);

export default ReduxHello;
