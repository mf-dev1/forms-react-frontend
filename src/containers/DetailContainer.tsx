import * as React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { ApplicationState } from '../reducers';
import { fetchDetail, fetchDetailTitle } from '../actions/detailActions';
import Detail from '../components/Detail';
import { DetailState } from '../reducers/detailReducer';
import { ConnectedReduxProps } from '../reducers/index';
import { RouteComponentProps } from 'react-router-dom';

interface PropsFromDispatch {
  fetchDetail: (id: string) => any;
  fetchDetailTitle: (id: string) => any;
}

interface RouteParams {
  id: string;
}

type AllProps = DetailState &
  PropsFromDispatch &
  RouteComponentProps<RouteParams> &
  ConnectedReduxProps;

class DetailContainer extends React.Component<AllProps> {
  public componentDidMount() {
    const id = this.props.match.params.id;
    this.props.fetchDetail(id);
    this.props.fetchDetailTitle(id);
  }

  public render() {
    const id = this.props.match.params.id;
    const title = this.props.meta ? this.props.meta.title : '';
    const items = this.props.data || [];

    if (this.props.loading) {
      return <div className="home-container">Loading...</div>;
    } else {
      return <Detail id={id} title={title} items={items} />;
    }
  }
}

const mapStateToProps = (state: ApplicationState) => {
  return {
    loading: state.detail.loading,
    meta: state.detail.meta,
    data: state.detail.data,
    errors: state.detail.errors,
  };
};

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      fetchDetail,
      fetchDetailTitle,
    },
    dispatch
  );
};

// Connected Component
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DetailContainer);
