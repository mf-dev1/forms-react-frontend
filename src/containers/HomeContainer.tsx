import * as React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { ApplicationState } from '../reducers';
import { fetchHome } from '../actions/homeActions';
import Home from '../components/Home';
import WithLoading from '../components/WithLoading';

import { HomeState } from '../reducers/homeReducer';
import { ConnectedReduxProps } from '../reducers/index';
import { RouteComponentProps } from 'react-router-dom';

interface PropsFromDispatch {
  fetchHome: typeof fetchHome;
}

type AllProps = HomeState &
  PropsFromDispatch &
  RouteComponentProps<{}> &
  ConnectedReduxProps;

class HomeContainer extends React.Component<AllProps> {
  public componentDidMount() {
    this.props.fetchHome();
  }

  public render() {
    const items = this.props.data || [];
    const loading = this.props.loading;
    const errors = this.props.errors;
    const HomeWithLoading = WithLoading(Home);

    return <HomeWithLoading loading={loading} errors={errors} items={items} />;
  }
}

const mapStateToProps = (state: ApplicationState) => {
  return {
    loading: state.home.loading,
    data: state.home.data,
    errors: state.home.errors,
  };
};

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      fetchHome,
    },
    dispatch
  );
};

// Connected Component
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HomeContainer);
