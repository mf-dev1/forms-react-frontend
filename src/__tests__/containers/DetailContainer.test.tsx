import * as React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import { DetailContainer } from '../../containers/DetailContainer';

describe('Detail container', () => {
  let shallow;
  const mockFetch = jest.fn();

  beforeEach(() => {
    shallow = createShallow();
  });

  it('renders without crashing', () => {
    const mockState = {
      loading: true,
      data: [],
      errors: undefined,
      meta: {},
      voted: false,
    };
    const mockProps = {
      loading: mockState.loading,
      data: mockState.data,
      errors: mockState.errors,
      voted: mockState.voted,
      fetchDetail: mockFetch,
      fetchDetailTitle: mockFetch,
      voteDetail: mockFetch,
    };
    const wrapper = shallow(<DetailContainer {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
