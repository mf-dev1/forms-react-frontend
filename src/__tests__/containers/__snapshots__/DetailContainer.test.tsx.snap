// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail container renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DetailContainer
    data={Array []}
    errors={undefined}
    fetchDetail={[Function]}
    fetchDetailTitle={[Function]}
    loading={true}
    voteDetail={[Function]}
    voted={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "errors": undefined,
      "id": "",
      "items": Array [],
      "loading": true,
      "onSave": [Function],
      "title": "",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "errors": undefined,
        "id": "",
        "items": Array [],
        "loading": true,
        "onSave": [Function],
        "title": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {},
  },
}
`;
